File: Backe/api/src/main/java/team/api/entity/Showtime.java
package team.api.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "showtimes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Showtime {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "showtime_id")
    private Integer showtimeId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id", nullable = false, foreignKey = @ForeignKey(name = "fk_show_movie"))
    private Movie movie;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false, foreignKey = @ForeignKey(name = "fk_show_room"))
    private Room room;

    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "base_price", nullable = false, precision = 10, scale = 2)
    private BigDecimal basePrice;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    @Column(name = "buffer_time_minutes")
    @Builder.Default
    private Integer bufferTimeMinutes = 15;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    @Column(columnDefinition = "enum('active', 'cancelled', 'completed') default 'active'")
    private Status status = Status.active;

    @Column(name = "batch_id", length = 100)
    private String batchId;

    public enum Status {
        active, cancelled, completed
    }
}

File: Backe/DotnetBackend/Entities/Showtime.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebPhimApi.Entities;

[Table("showtimes")]
public class Showtime
{
    [Key]
    [Column("showtime_id")]
    public int ShowtimeId { get; set; }

    [Column("movie_id")]
    public int MovieId { get; set; }

    [Column("room_id")]
    public int RoomId { get; set; }

    [Required]
    [Column("start_time")]
    public DateTime StartTime { get; set; }

    [Column("end_time")]
    public DateTime? EndTime { get; set; }

    [Column("buffer_time_minutes")]
    public int BufferTimeMinutes { get; set; } = 15;

    [Required]
    [Column("base_price", TypeName = "decimal(10,2)")]
    public decimal BasePrice { get; set; }

    [Column(TypeName = "enum('active','cancelled','completed')")]
    public ShowtimeStatus Status { get; set; } = ShowtimeStatus.active;

    [MaxLength(100)]
    [Column("batch_id")]
    public string? BatchId { get; set; }

    // Navigation
    [ForeignKey(nameof(MovieId))]
    public Movie Movie { get; set; } = null!;

    [ForeignKey(nameof(RoomId))]
    public Room Room { get; set; } = null!;

    public ICollection<Booking> Bookings { get; set; } = [];
    public ICollection<BookingSeat> BookingSeats { get; set; } = [];
}

public enum ShowtimeStatus { active, cancelled, completed }
